src/SoftRobots/component/controller/CommunicationController.inl

```diff
@@ -318,22 +318,22 @@ void CommunicationController<DataTypes>::sendData()
 template<class DataTypes>
 void CommunicationController<DataTypes>::receiveData()
 {
-    if(d_pattern.getValue().getSelectedItem() == "request/reply")
-        sendRequest();
+	if (d_pattern.getValue().getSelectedItem() == "request/reply")
+		sendRequest();
 
     zmq::message_t message;
-    unsigned int messageSize;
     bool status = m_socket->recv(&message);
+
     if(status)
     {
-        char messageChar[messageSize];
-        memcpy(&messageChar, message.data(), messageSize);
+        char* messageChar = new char[message.size()];
```

Comment:
Isn't there a memory leak here ? I don't see the corresponding delete.

Reply:
Good catch, @damienmarchal ! I think I fixed it.